dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -53,7 +53,10 @@ export class FrontendGenerator extends AbstractGenerator {
 </head>
 
 <body>
-  <div class="theia-preload">${this.compileIndexPreload(frontendModules)}</div>
+  <div class="theia-preload">
+    ${this.compileIndexPreload(frontendModules)}
+    <i class="codicon codicon-sync theia-loader"></i>
```

Comment:
It will break all clients having custom index preload?

Reply:
@akosyakov how can we avoid this?

Reply:
Probably we can use a pseudo-element of `theia-preload`, but we'll have to duplicate codicon's style setting its background. Will try to do so.

Reply:
Works fine, the only problem is that bad character is displayed a bit of time while font is being loaded.

Reply:
There is a code above which returns if the template is missing. So far it was returning an empty string, you could return something else there. I am not sure that adding/changing styles is good.

Reply:
Which style exactly? Where's that code?